### ISSUE TITLE
PayPal Auth with webview

### ISSUE THREAD

**Issue by antons81** [paypal/PayPal-iOS-SDK#408]
I am building an app with PalPal integration. What I need is Authentication through WebView with PayPal login screen. So user will enter his credentials, I will get a token and get his email account to save in my app. Something like Instagram.

What's wrong there, because I cannot understand anything from PayPal off docs

webView = UIWebView(frame: self.view.bounds)
webView.autoresizingMask = [UIViewAutoresizing.FlexibleHeight, UIViewAutoresizing.FlexibleHeight]

    let url = "https://api.sandbox.paypal.com/v1/oauth2/token/?client_id=\(client_id)&secret=\(secret)&grant_type=client_credentials&response_type=token"

    webView.delegate = self
    //webView.scalesPageToFit = true
    webView.loadRequest(NSURLRequest(URL: NSURL(string: url)!))
    self.view.addSubview(webView)
But givin an error:

{
"error" : "invalid client", 
"error description" : "client credentials are missing"
}

In other words I need to show to user username and password fields from paypal, than get token and take his verified paypal email to save in my app.

**Comment by bluk** [comment]
The flow you're describing is not supported. You should not embed your secret in "insecure" clients (the code could be reverse engineered, and your secret could be stolen which could lead to malicious users using your client ID and secret doing bad things).

You should look into [Profile Sharing](https://github.com/paypal/PayPal-iOS-SDK/blob/master/docs/profile_sharing_mobile.md) from this mSDK if you want more information about a specific user via a native integration. The other flows (single payment, future payments) are also available for making payments (e.g. if you want to get an authorization for future payments so you can charge an account without asking the user to re-login to PayPal).

If these features don't satisfy your use case, can you leave a comment and elaborate on your use case? Thanks.

**Comment by antons81** [comment]
Hello Bluk, thanks for your answer. Ok if user will add his Paypal as payment option. How I have add it? Just give him input his paypal email? Doesn't matter if it's real or not real?

**Comment by bluk** [comment]
If you can check out [this repository's sample app](https://github.com/paypal/PayPal-iOS-SDK/tree/master/SampleApp), you can get an idea of how you can integrate with this mobile SDK and the flows on what is provided (e.g. how the customer would enter in their credentials and authorize a payment). There is also a [sample in Swift](https://github.com/paypal/PayPal-iOS-SDK/tree/master/SampleApp-Swift) since that seems to be the language you like to use.

Another option is to look at Braintree (a PayPal owned company) and see about the [Braintree iOS SDK](https://github.com/braintree/braintree_ios) depending if Braintree is supported in your country.
